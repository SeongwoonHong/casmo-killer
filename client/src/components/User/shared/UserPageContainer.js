import React from 'react';

import LoadingOverlay from 'sharedComponents/LoadingOverlay';

const UserPageContainer = (props) => {
  return (
    <div className={ props.className }>
      {
        props.title
          ? (
            <h2 className="user-page-title">
              { props.title }
              <i className="material-icons">
                { props.icon }
              </i>
            </h2>
          )
          : null
      }
      <div className="user-form-box">
        <LoadingOverlay
          isVisible={ props.isLoading }
          isComplete={ props.isComplete }
          overlayColor="rgba(256,256,256,.75)"
          circleColor="#1F4B40" />
        <div className="user-form-header">
          <h3>{ props.formTitle }</h3>
          {
            props.formMsg
             ? <p>{ props.formMsg }</p>
             : null
          }
        </div>
        <form
          noValidate
          className="user-form"
          onSubmit={ (e) => {
            e.preventDefault();
            props.onSubmit();
          } }>
          { props.children }
          {
            props.button
              ? props.button
              : (
                <button
                  type="submit"
                  className="user-form-button"
                  disabled={ props.disabled }>
                  Submit
                </button>
              )
          }
        </form>
      </div>
    </div>
  );
};

export default UserPageContainer;
