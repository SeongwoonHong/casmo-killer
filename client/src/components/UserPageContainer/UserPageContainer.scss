.User {
    &__page__title {
        @include flex-box(space-between);
        font-size: 16px;
        color: palette(Grey, 900);
        margin: $user-box-padding $user-box-padding - 5px;
        font-weight: 800;
        text-transform: uppercase;
        &__icons {
            border: 1px solid palette(Grey, 900);
            border-radius: 5px;
            font-size: 16px;
            padding: 3px;
        }
    }
    &__form {
        &__row {
            background-color: #fff;
            border: 1px solid $main-menu-border-color;
            margin-bottom: 10px;
        }
        &__header {
            border-bottom: 1px solid $main-menu-border-color;
            color: palette(Grey, 800);
            padding: $user-box-padding;
            &__title {
                font-size: 14px;
                font-weight: 700;
            }
            &__subtitle {
                margin-top: $user-box-padding;
            }
        }
        &__btn {
            background-color: palette(Grey, 700);
            color: palette(Grey, 50);
            display: block;
            font-size: 12px;
            height: 30px;
            line-height: 30px;
            padding: 0 10px;
            &:disabled {
                background-color: palette(Grey, 300);
            }
            &:last-child {
                margin: $user-box-padding $user-box-padding $user-box-padding auto;
            }
        }
    }
}
