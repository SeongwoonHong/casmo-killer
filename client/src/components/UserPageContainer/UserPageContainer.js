import React from 'react';
import PropTypes from 'prop-types';

import LoadingOverlay from '@sharedComponents/LoadingOverlay';

import './UserPageContainer.scss';

const UserPageContainer = (props) => {

  return (
    <div className={ props.className }>
      {
        props.title
          ? (
            <h2 className="User__page__title">
              { props.title }
              <i className="User__page__title__icons material-icons">
                { props.icon }
              </i>
            </h2>
          )
          : null
      }
      <div className={ `User__form__row ${props.className}__form__row`}>
        <LoadingOverlay
          isVisible={ props.isLoading }
          overlayColor="rgba(256,256,256,.75)"
          circleColor="#1F4B40" />
        <div className={ `User__form__header ${props.className}__form__header`}>
          <h3 className={ `User__form__header__title ${props.className}__form__header__title`}>
            { props.formTitle }
          </h3>
          {
            props.formMsg
              ? (
                <p className={ `User__form__header__subtitle ${props.className}__form__header__subtitle` }>
                  { props.formMsg }
                </p>
              )
              : null
          }
        </div>
        <form
          noValidate
          className={ `User__form ${props.className}__form user-form` }
          onSubmit={ (e) => {
            e.preventDefault();
            props.onSubmit();
          } }>
          { props.children }
          {
            props.button
              ? props.button
              : (
                <button
                  type="submit"
                  className="User__form__btn user-form-button"
                  disabled={ props.disabled }>
                  Submit
                </button>
              )
          }
        </form>
      </div>
    </div>
  );

};

UserPageContainer.propTypes = {
  className: PropTypes.string.isRequired,
  title: PropTypes.string,
  icon: PropTypes.string,
  isLoading: PropTypes.bool,
  formTitle: PropTypes.string.isRequired,
  formMsg: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.any.isRequired,
  disabled: PropTypes.bool
};

UserPageContainer.defaultProps = {
  title: '',
  icon: '',
  isLoading: false,
  formMsg: '',
  disabled: false
};

export default UserPageContainer;
