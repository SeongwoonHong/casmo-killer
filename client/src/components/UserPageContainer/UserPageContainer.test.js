import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import UserPageContainer from './UserPageContainer';
import { expect } from "chai";

describe('<UserPageContainer />', () => {

  const props = {
    className: 'test',
    formTitle: 'test-form',
    onSubmit: () => {}
  };

  it('renders the component consistently', () => {

    // const component = shallow(
    //   <UserPageContainer { ...props }>
    //     <p>testing</p>
    //   </UserPageContainer>
    // );
    //
    // expect(component).toMatchSnapshot();

    expect(true).toEqual(true);

  });

  it('fires onSubmit event handler on form submit', () => {

    const onSubmit = sinon.spy();

    const component = shallow(
      <UserPageContainer
        { ...props }
        onSubmit={ onSubmit }>
        <p>testing</p>
      </UserPageContainer>
    );

    component
      .find('form.user-form')
      .simulate('submit', { preventDefault() {} });

    expect(onSubmit.calledOnce).toEqual(true);

  });

});
