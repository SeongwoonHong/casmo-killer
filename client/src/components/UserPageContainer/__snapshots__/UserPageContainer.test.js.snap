// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPageContainer /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserPageContainer
    className="test"
    disabled={false}
    formMsg=""
    formTitle="test-form"
    icon=""
    isLoading={false}
    onSubmit={[Function]}
    title=""
>
    <p>
        testing
    </p>
</UserPageContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="User__form__row test__form__row"
>
          <LoadingOverlay
                    circleColor="#1F4B40"
                    circleSize=""
                    isVisible={false}
                    overlayColor="rgba(256,256,256,.75)"
          />
          <div
                    className="User__form__header test__form__header"
          >
                    <h3
                              className="User__form__header__title test__form__header__title"
                    >
                              test-form
                    </h3>
          </div>
          <form
                    className="User__form test__form user-form"
                    noValidate={true}
                    onSubmit={[Function]}
          >
                    <p>
                              testing
                    </p>
                    <button
                              className="User__form__btn test__form__btn"
                              disabled={false}
                              type="submit"
                    >
                              Submit
                    </button>
          </form>
</div>,
      ],
      "className": "test",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <LoadingOverlay
              circleColor="#1F4B40"
              circleSize=""
              isVisible={false}
              overlayColor="rgba(256,256,256,.75)"
/>,
            <div
              className="User__form__header test__form__header"
>
              <h3
                            className="User__form__header__title test__form__header__title"
              >
                            test-form
              </h3>
</div>,
            <form
              className="User__form test__form user-form"
              noValidate={true}
              onSubmit={[Function]}
>
              <p>
                            testing
              </p>
              <button
                            className="User__form__btn test__form__btn"
                            disabled={false}
                            type="submit"
              >
                            Submit
              </button>
</form>,
          ],
          "className": "User__form__row test__form__row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "circleColor": "#1F4B40",
              "circleSize": "",
              "isVisible": false,
              "overlayColor": "rgba(256,256,256,.75)",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="User__form__header__title test__form__header__title"
>
                  test-form
</h3>,
                null,
              ],
              "className": "User__form__header test__form__header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test-form",
                  "className": "User__form__header__title test__form__header__title",
                },
                "ref": null,
                "rendered": "test-form",
                "type": "h3",
              },
              null,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  testing
</p>,
                <button
                  className="User__form__btn test__form__btn"
                  disabled={false}
                  type="submit"
>
                  Submit
</button>,
              ],
              "className": "User__form test__form user-form",
              "noValidate": true,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "testing",
                },
                "ref": null,
                "rendered": "testing",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "User__form__btn test__form__btn",
                  "disabled": false,
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="User__form__row test__form__row"
>
            <LoadingOverlay
                        circleColor="#1F4B40"
                        circleSize=""
                        isVisible={false}
                        overlayColor="rgba(256,256,256,.75)"
            />
            <div
                        className="User__form__header test__form__header"
            >
                        <h3
                                    className="User__form__header__title test__form__header__title"
                        >
                                    test-form
                        </h3>
            </div>
            <form
                        className="User__form test__form user-form"
                        noValidate={true}
                        onSubmit={[Function]}
            >
                        <p>
                                    testing
                        </p>
                        <button
                                    className="User__form__btn test__form__btn"
                                    disabled={false}
                                    type="submit"
                        >
                                    Submit
                        </button>
            </form>
</div>,
        ],
        "className": "test",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <LoadingOverlay
                circleColor="#1F4B40"
                circleSize=""
                isVisible={false}
                overlayColor="rgba(256,256,256,.75)"
/>,
              <div
                className="User__form__header test__form__header"
>
                <h3
                                className="User__form__header__title test__form__header__title"
                >
                                test-form
                </h3>
</div>,
              <form
                className="User__form test__form user-form"
                noValidate={true}
                onSubmit={[Function]}
>
                <p>
                                testing
                </p>
                <button
                                className="User__form__btn test__form__btn"
                                disabled={false}
                                type="submit"
                >
                                Submit
                </button>
</form>,
            ],
            "className": "User__form__row test__form__row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "circleColor": "#1F4B40",
                "circleSize": "",
                "isVisible": false,
                "overlayColor": "rgba(256,256,256,.75)",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="User__form__header__title test__form__header__title"
>
                    test-form
</h3>,
                  null,
                ],
                "className": "User__form__header test__form__header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "test-form",
                    "className": "User__form__header__title test__form__header__title",
                  },
                  "ref": null,
                  "rendered": "test-form",
                  "type": "h3",
                },
                null,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    testing
</p>,
                  <button
                    className="User__form__btn test__form__btn"
                    disabled={false}
                    type="submit"
>
                    Submit
</button>,
                ],
                "className": "User__form test__form user-form",
                "noValidate": true,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "testing",
                  },
                  "ref": null,
                  "rendered": "testing",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "User__form__btn test__form__btn",
                    "disabled": false,
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
